### ISSUE TITLE
[@typescript-eslint/typedef] false positive when using array destructing on function arguments

### ISSUE THREAD

**Issue by Ricky-rick** [typescript-eslint/typescript-eslint#2433]
False positive when using array destructing on function arguments

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**
run `yarn eslint` with the code and config below  

**Config**
```JSON
{
    "rules": {
      "@typescript-eslint/typedef": [
            "error",
            {
              "arrayDestructuring": true,
            }
          ]
    }
}
```
**Code**
_(I know this is a silly example but it's the most simple explanation)_
```TS
function sumStrLen(...[str1, str2]: string[]): number {
  return str1.length + str2.length
}

sumStrLen('foo', 'bar')
```

**Expected Result**

I dit expect no error because i typed the destructed array as string[] 

**Actual Result**

I got an lint-error:`Expected a type annotation  @typescript-eslint/typedef`

**Additional Info**
This is a workaround that did work
```TS
function sumStrLen(...string: string[]): number {
  const [str1, str2]: string[] = string
  return str1.length + str2.length
}

sumStrLen('foo', 'bar')
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.10.1` |
| `@typescript-eslint/parser`        | `3.10.1` |
| `TypeScript`                       | `4.0.2` |
| `ESLint`                           | `7.7.0` |
| `node`                             | `10.22.0` |

**Comment by bradzacher** [comment]
The bug here is that a rest parameter owns the type annotation, not the array pattern.
Happy to accept a PR
